op-node/rollup/derive/l1_block_info.go

```diff
@@ -335,3 +366,53 @@ func L1InfoDepositBytes(rollupCfg *rollup.Config, sysCfg eth.SystemConfig, seqNu
 	}
 	return opaqueL1Tx, nil
 }
+
+func DepositsCompleteDeposit(rollupCfg *rollup.Config, sysCfg eth.SystemConfig, seqNumber uint64, block eth.BlockInfo, l2BlockTime uint64) (*types.DepositTx, error) {
+	l1BlockInfo := L1BlockInfo{
+		Number:         block.NumberU64(),
+		Time:           block.Time(),
+		BaseFee:        block.BaseFee(),
+		BlockHash:      block.Hash(),
+		SequenceNumber: seqNumber,
+		BatcherAddr:    sysCfg.BatcherAddr,
+	}
+	data, err := l1BlockInfo.marshalDepositsComplete()
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal Bedrock l1 block info: %w", err)
+	}
+	source := L1InfoDepositSource{
+		L1BlockHash: block.Hash(),
+		SeqNumber:   seqNumber,
+	}
+	// Set a normal gas limit with `IsSystemTransaction` to ensure
+	// that the DepositCompleted Transaction does not run out of gas.
+	out := &types.DepositTx{
+		SourceHash:          source.SourceHash(),
+		From:                L1InfoDepositerAddress,
+		To:                  &L1BlockAddress,
+		Mint:                nil,
+		Value:               big.NewInt(0),
+		Gas:                 50_000,
+		IsSystemTransaction: false,
+		Data:                data,
```

Comment:
The data here should be simply `DepositsCompleteBytes4` and we don't need the `l1BlockInfo` struct here